### ISSUE TITLE
1.2.0.dev20190723 Conv1D Illegal instruction (core dumped)

### ISSUE THREAD

**Issue by jamarshon** [pytorch/pytorch#23231]
## Setup
```
conda create --name env3.6 python=3.6
conda activate env3.6
conda install numpy ninja pyyaml mkl mkl-include setuptools cmake cffi typing
conda install pytorch-nightly-cpu -c pytorch
```
Installs 1.2.0.dev20190723

## Code
```
import torch

conv_stride = 2
num_channels = 1
wave_to_conv = torch.rand((1, 1, 5024))
weights = torch.rand(1, 1, 25)
conv = torch.nn.functional.conv1d(wave_to_conv, weights, stride=conv_stride)
```

## Output
```
 Illegal instruction (core dumped)
```
Previous versions worked
```
>>> torch.__version__
'1.2.0.dev20190615'
>>> conv.shape
torch.Size([1, 1, 2500])
```

## Trace
```
#0  0x00007fffe71367b8 in mkldnn::impl::memory_desc_wrapper::compute_blocking(mkldnn_memory_desc_t&) ()
   from /private/home/jamarshon/anaconda3/envs/env3.6/lib/python3.6/site-packages/torch/lib/libtorch.so
#1  0x00007fffe7130c0e in mkldnn_memory_desc_init () from /private/home/jamarshon/anaconda3/envs/env3.6/lib/python3.6/site-packages/torch/lib/libtorch.so
#2  0x00007fffe4e0858c in at::native::itensor_view_from_dense(at::Tensor const&) ()
   from /private/home/jamarshon/anaconda3/envs/env3.6/lib/python3.6/site-packages/torch/lib/libtorch.so
#3  0x00007fffe4df1540 in at::native::mkldnn_convolution(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, long) () from /private/home/jamarshon/anaconda3/envs/env3.6/lib/python3.6/site-packages/torch/lib/libtorch.so
#4  0x00007fffe4fd312f in at::TypeDefault::mkldnn_convolution(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, long) () from /private/home/jamarshon/anaconda3/envs/env3.6/lib/python3.6/site-packages/torch/lib/libtorch.so
#5  0x00007fffe66a4654 in torch::autograd::VariableType::mkldnn_convolution(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, long) () from /private/home/jamarshon/anaconda3/envs/env3.6/lib/python3.6/site-packages/torch/lib/libtorch.so
#6  0x00007fffe4a160eb in at::mkldnn_convolution(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, long) ()
   from /private/home/jamarshon/anaconda3/envs/env3.6/lib/python3.6/site-packages/torch/lib/libtorch.so
#7  0x00007fffe4a1ffd7 in at::native::_convolution(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long, bool, bool, bool) () from /private/home/jamarshon/anaconda3/envs/env3.6/lib/python3.6/site-packages/torch/lib/libtorch.so
#8  0x00007fffe4fcf00f in at::TypeDefault::_convolution(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long, bool, bool, bool) () from /private/home/jamarshon/anaconda3/envs/env3.6/lib/python3.6/site-packages/torch/lib/libtorch.so
#9  0x00007fffe6708dc2 in torch::autograd::VariableType::_convolution(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long, bool, bool, bool) () from /private/home/jamarshon/anaconda3/envs/env3.6/lib/python3.6/site-packages/torch/lib/libtorch.so
#10 0x00007fffe4a16c8b in at::native::convolution(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, c10::ArrayRef<long>, long) () from /private/home/jamarshon/anaconda3/envs/env3.6/lib/python3.6/site-packages/torch/lib/libtorch.so
```

## Environment
PyTorch version: 1.2.0.dev20190723
Is debug build: No
CUDA used to build PyTorch: Could not collect

OS: Ubuntu 18.04.1 LTS
GCC version: (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
CMake version: version 3.14.0

Python version: 3.6
Is CUDA available: No
CUDA runtime version: 10.0.130
GPU models and configuration:
GPU 0: Quadro GP100
GPU 1: Quadro GP100

Nvidia driver version: 410.79
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.16.4
[pip] torch==1.2.0.dev20190723
[pip] torchaudio==0.2
[conda] blas                      1.0                         mkl
[conda] mkl                       2019.4                      243
[conda] mkl-include               2019.4                      243
[conda] mkl_fft                   1.0.12           py36ha843d7b_0
[conda] mkl_random                1.0.2            py36hd81dba3_0
[conda] pytorch-nightly-cpu       1.2.0.dev20190723     py3.6_cpu_0    pytorch
[conda] torchaudio                0.2                       dev_0    <develop>

**Comment by jamarshon** [comment]
Looking for something merged on 07-22/23. 
https://github.com/pytorch/pytorch/pulls?q=is%3Apr+is%3Aclosed+label%3Amerged+closed%3A2019-07-22..2019-07-23+base%3Amaster+sort%3Aupdated-desc

**Comment by jamarshon** [comment]
Looks like it works when building from source.
```
1.2.0a0+60c46dd
torch.Size([1, 1, 2500])
```
Must have been an issue with conda nightly

**Comment by soumith** [comment]
@jamarshon can you specify what CPU you have? you can get it by doing `cat /proc/cpuinfo`.

@gujinghui @Jianhui-Li @uyongw @jgong5 @XiaobingSuper could you help figure out why the MKLDNN code is being compiled with `-mnative`, I suspect that's the main issue -- i.e. MKL-DNN is being compiled on a machine with AVX2, but then @jamarshon's machine doesn't have AVX2 support.

**Comment by soumith** [comment]
this is a blocker for the 1.2 release next week

**Comment by vpirogov** [comment]
@soumith, Intel MKL-DNN [default behavior](http://intel.github.io/mkl-dnn/dev_guide_build_options.html) is to compile code with -mnative for GCC 5>=5.0. This behavior can be overriden with build option `MKLDNN_ARCH_OPT_FLAGS=""`.

We will be changing this behavior in v1.1 to avoid portability issues as this one.

**Comment by jamarshon** [comment]
[cpu.txt](https://github.com/pytorch/pytorch/files/3423215/cpu.txt)
```
model name	: Intel(R) Xeon(R) CPU E5-2698 v4 @ 2.20GHz
```
It's worth noting it has the same message on torchaudio's [CI](https://travis-ci.org/pytorch/audio/jobs/562594119) which is Ubuntu 16.04.6 LTS xenial

**Comment by soumith** [comment]
@vpirogov if we provide `MKLDNN_ARCH_OPT_FLAGS` manually, would that also affect the kernels that it compiles?

The situation I am hoping for is:
- MKLDNN_ARCH_OPT_FLAGS="-msse -msse2 -msse3"
- the high performance kernels are still using avx / avx2 via runtime-dispatch

**Comment by vpirogov** [comment]
High performance kernels use just-in-time assembly and detect ISA in the runtime using CPUID. These are not dependent on compiler (or it's optimization flags). 

There are some implementations that use C that might be impacted by compiler flags, we don't expect the impact to be significant though.

It's should be safe to use `MKLDNN_ARCH_OPT_FLAGS="-msse4"` for the systems produced in the last 10 years.

**Comment by gujinghui** [comment]
fixed in #23292 

@soumith
@vpirogov
@Jianhui-Li
@jgong5
@uyongw

**Comment by gchanan** [comment]
closing this as #23292 merged.
